### PR TITLE
Link space type to facility

### PR DESCRIPTION
**Pull request by mathias234**
Gjenstår før vi tar inn i main:

- [x] Seed for spacetype <> facility links for fasilitetene + space typene våre@
- [x] UI improvements for facility#edit
- [x] Search respecting link between facility <> spacetype

### CODE REVIEW

**Comment by DanielJackson-Oslo** [comment]
Si fra når den er klar for review, her klør det i fingra 😂

**Comment by mathias234** [comment]
> Si fra når den er klar for review, her klør det i fingra joy

Tror den er klar nå

**Review by DanielJackson-Oslo — CHANGES_REQUESTED** [review]
Basic structure seems to work, but at first there were some things I couldn't get working.

While reviewing, to understand the problems, I tried to fix them in #187 Not sure if that's useful code or not, but then at least you have it too!

- FacilityReview#new view file crashed on load, with or without space type <> facility links set up. At first I thought it might be because it doesn't check for `@not_matching_space_types` and added a check for that, but it turns out it was because `not_matching_space_types` and `matching_space_types` where not included in the controller for #new!  Fixed in #187 

- With no space_type <> Facility link present, no facilities are shown at all. I don't think that's good behaviour, but it can be fixed in a future PR.

- The join on line 153 in space model seems to filter Facilities on SpaceType too early, so that the function for returning Facilities that are not filtered on that Spacetype did not work.

File: app/controllers/facility_reviews_controller.rb [hunk 1]
```diff
@@ -61,6 +62,26 @@ def create_success
 
   private
 
+  def filter_matching_space_types
```
Comment (DanielJackson-Oslo) [inline app/controllers/facility_reviews_controller.rb:65]: These two functions are never called in `new`, and so are unused.

Comment (mathias234) [reply app/controllers/facility_reviews_controller.rb:65]: Ah I probably extracted those to functions without testing (probably because of RuboCop 😆)

Comment (DanielJackson-Oslo) [reply app/controllers/facility_reviews_controller.rb:65]: Rubocop strikes again!

File: app/models/space.rb [hunk 2]
```diff
@@ -144,18 +144,32 @@ def reviews_for_categories(user)
...
-      .includes(facility: [:facilities_categories])
-      .joins(facility: [:facilities_categories])
-      .each_with_object({}) do |space_facility, memo|
+  #
+  # if match_all_except_unknown is true it will give you all the facilities that match the current space's space_types
+  # and all the "other" facilities where experience is not unknown
+  def facilities_for_categories(match_all_except_unknown: false) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+    result = space_facilities
+             .includes(facility: [:facilities_categories, :space_types])
+             .joins(facility: [:facilities_categories, :space_types])
```
[the remark this answers is not in the document]
Comment (mathias234) [reply app/models/space.rb:153]: See comment in #187 for more info

**Review by DanielJackson-Oslo — APPROVED** [review]
Approved as soon as the SpaceType-linked search is merged. This is a GREAT change and vastly improves the default experience of the app, as we can give a lot of useful information from just knowing what space type somethng is.

Good work!

**Comment by DanielJackson-Oslo** [comment]
> Approved as soon as the SpaceType-linked search is merged. This is a GREAT change and vastly improves the default experience of the app, as we can give a lot of useful information from just knowing what space type somethng is.
> 
> Good work!

@mathias234 This is now ready to merge and get into production!

**Pull request merged**